### PR TITLE
MM-48753 - Calls: GA requirements

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- This PR addresses the move to GA requirements for mobile from https://docs.google.com/document/d/1Tr8OBH4nPSN3SIMXWKY-_TMV-C0ht5zxmc2rLthZlLA/edit .
- It requires a corresponding GA PR from the calls repo, which isn't submitted yet. But the changes here are backwards compatible.
- I've gone through all the test cases outlined this sheet (don't use the tables in the doc, the linking isn't working reliably): https://docs.google.com/spreadsheets/d/1aRc_zq2gaLLjTTx6ua1KwVfFRGD51xLYsNRRG5MUIH0/edit#gid=0 )
- There are three combinations of App (new x old) x Server (new x old), ignoring the case of App old - Server old. For each of those, you need to test x iOS & Android. For each cell in the test cases you have to test the expected outcome and the other possibilities (the negative outcomes). (I will need to write up a testing plan for this, it's a little involved.)

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-48753

#### Checklist
- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: 

#### Release Note
```release-note
Calls: Adjusted how system console settings are interpreted in the Calls UI
```

### CODE REVIEW

**Review by enahum — APPROVED** [review]
Lgtm

File: app/products/calls/alerts.ts [hunk 1]
```diff
@@ -113,6 +115,40 @@ const doJoinCall = async (serverUrl: string, channelId: string, isDMorGM: boolea
...
+            // if GA calls:
+            //   if explicitly enabled, everyone can start a call
+            //   if !explicitly enabled and defaultEnabled, everyone can start
+            //   if !explicitly enabled and !defaultEnabled, system admins can start, regular users get alert
+            // Note: the below is a 'badly' coded if. But it's clear, which trumps.
+            if (enabled || (!enabled && DefaultEnabled) || (!enabled && !DefaultEnabled && isAdmin)) {
+                if (isAdmin) {
+                    const config = getCallsConfig(serverUrl);
+
+                    await sendEphemeralPost(
```
[the remark this answers is not in the document]
Comment (cpoile) [reply app/products/calls/alerts.ts:136]: Yep, everytime while Calls is in "testing" mode.

**Review by streamer45 — APPROVED** [review]
Thanks @cpoile , very thorough :+1:  Just left a couple of non-blocking questions.

File: app/products/calls/alerts.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (cpoile) [reply app/products/calls/alerts.ts]: Ah, good point. Fixed by moving it to the server. Thanks!

File: app/products/calls/alerts.ts [hunk 2]
```diff
@@ -113,6 +115,40 @@ const doJoinCall = async (serverUrl: string, channelId: string, isDMorGM: boolea
...
         }
+
+        if (newCall) {
+            const enabled = getCallsState(serverUrl).enabled[channelId];
+            const {DefaultEnabled} = getCallsConfig(serverUrl);
+            const isAdmin = isSystemAdmin(user.roles);
+
+            // if explicitly disabled, we wouldn't get to this point.
+            // if pre-GA calls:
+            //   we won't get here
```
[the remark this answers is not in the document]
Comment (cpoile) [reply app/products/calls/alerts.ts]: Ah, I forgot to update that comment. Fixed. We do get here, but it's simpler.

File: app/products/calls/alerts.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (cpoile) [reply app/products/calls/alerts.ts]: That's a much better solution, let's do that. And this was wrong anyway (pointed out by Elias above).

File: app/products/calls/observers/index.ts [hunk 3]
```diff
@@ -1,38 +1,41 @@
...
-        distinctUntilChanged(),
+    const callsStateEnabledDict = observeCallsState(serverUrl).pipe(
+        switchMap((state) => of$(state.enabled)),
+        distinctUntilChanged(), // Did the enabled object ref change? If so, a channel's enabled state has changed.
     );
-    const isDMChannel = observeChannel(database, channelId).pipe(
-        switchMap((c) => of$(c?.type === General.DM_CHANNEL)),
-        distinctUntilChanged(),
+    const callsGAServer = observeConfigValue(database, 'Version').pipe(
+        switchMap((v) => of$(isMinimumServerVersion(v || '', 7, 6))),
```
[the remark this answers is not in the document]
Comment (cpoile) [reply app/products/calls/observers/index.ts:27]: That's the assumption I'm making -- do you think it's ok? @streamer45

[the remark this answers is not in the document]
Comment (streamer45) [reply app/products/calls/observers/index.ts:27]: Yes, it's totally fair. Mainly want us to be aware of the potential super edge case.

[the remark this answers is not in the document]
Comment (cpoile) [reply app/products/calls/observers/index.ts:27]: @streamer45 Right -- if someone loads an earlier version of calls manually. Imo, let's not bend over backwards to support that. :)

**Pull request merged**